src/aiovodafone/api.py

```diff
@@ -420,6 +453,300 @@ async def logout(self) -> None:
         _LOGGER.debug("Logout")
         await self._post_page_result("/api/v1/session/logout", payload={})
 
+    async def _post_page_result_with_handling(
+        self,
+        endpoint: str,
+        payload: dict,
+        timeout: int = 10,
+    ) -> aiohttp.ClientResponse | None:
+        """Wrap `_post_page_result` to handle errors and logging.
+
+        Args:
+        ----
+            endpoint (str): API endpoint for the POST request.
+            payload (dict): Data to be sent in the POST request.
+            timeout (int): Timeout for the request (default: 10 seconds).
+
+        Returns:
+        -------
+            aiohttp.ClientResponse | None: The response object if successful,
+            None otherwise.
+
+        """
+        try:
+            return await self._post_page_result(endpoint, payload, timeout)
+        except aiohttp.ClientResponseError:
+            _LOGGER.exception("Request to %s failed", endpoint)
+        return None
```

Comment:
Should we handle that in the common API ? Any drawback ?

Reply:
Don't see any, just wanted to make this PR the least invasive possible, since I can't test against Sercomm devices.